elyra/pipeline/validate.py

```diff
@@ -431,34 +432,36 @@ def _validate_environmental_variables(self, node_id: str, node_label: str,
                                        "propertyName": 'env_vars',
                                        "value": env_var})
 
-    def _validate_label(self, node_id: str, node_label: str, response: ValidationResponse) -> None:
+    def _validate_label(self, node_id: str, filename: str, node_label: str, response: ValidationResponse) -> None:
         """
         KFP specific check for the label name when constructing the node operation using dsl
         :param node_id: the unique ID of the node
+        :param filename: the name of the file with or without a relative path prefix
         :param node_label: the given node name or user customized name/label of the node
         :param response: ValidationResponse containing the issue list to be updated
         """
         label_name_max_length = 63
-        label_regex = re.compile('^[a-z0-9]([-a-z0-9]{0,62}[a-z0-9])?')
+        label_regex = re.compile('^[a-z0-9]([-_a-z0-9]{0,62}[a-z0-9])?')
```

Comment:
ack, I must have gotten some bad info during my research, will update the regex and update the tests.

Reply:
ok - I guess I'm not saying this is wrong per-se, just that it was interesting that it wouldn't have to also conform.  Given the label will typically derive from the filename for generic components and the commonality of underscores in filenames, I suspect the underscore is a valid label character.  I'd say if you find it works (and that character is preserved) then go for it.

Reply:
between out kfp processor, the kfp compiler, and the executefileop class all sanitizing the label/name to ultimately get a "good" name they can use, I am wondering if we should make these warnings instead of errors, since we do a lot of the name massaging behind the scenes, we can let them know that. We can move these regex requirements up into the validation logic, but would that be too harsh?   

Findings:
First is the distinguishing between the kubernetes pod label vs the kubeflow pipeline node name/display name. 

node label -> operation.name - node label is assigned to operation.name

operation.name -> sanitized_name in kfp processor, allows for max length to be over 63 chars and allows for letters, nums and dashes, this is used as the display name of the operation in kfp. I actually think that we might not need our own sanitize function anymore since the compiler sanitized a string for me after I explicitly disabled our method in testing.
![image](https://user-images.githubusercontent.com/13340013/129088717-eba546d6-74d7-4726-a954-e5ce4f975b51.png)

sanitized_name -> normalize_name in generic operators where max length is trimmed down to 63 and allows for alphanumerics[a-z0-9A-Z], dashes, underscores and dots. These values are actually used as k8s pod labels
![image](https://user-images.githubusercontent.com/13340013/129096403-22a7c819-a963-4157-b73d-215d4203bd9c.png) 

So a pod label would ultimately only have alphanumerics with dashes, since underscores and dots would have been removed by the sanitized_name method.

Reply:
I see the same thing (underscores promoted to dashes) in my testing.  Since we perform the sanitization, I would agree that we don't need strict validation.  It would be nice to issue an informational message indicating the new "sanitized name" when necessary.

Unless we want to make node-label required (which it sounds like it is), I'm not sure there's any validation to perform actually.

All validation was also being skipped when the node label derives from the filename as well (which is why I asked for the test), but that would be moot as well.